From 667c190fc08b8ef95bfa93b0516bcc58766ec0d1 Mon Sep 17 00:00:00 2001
From: Uri Eliyahu <uriel@marvell.com>
Date: Thu, 26 Dec 2013 19:07:59 +0200
Subject: [PATCH 1267/1825] pp2: prs sysfs tree rearrange

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit ba2411c1c3289e0b18254931ce972b462f91ba98

	- rename prsHigh directory to prs
	- remove parse low level api to pp2/prs/debug

Change-Id: I78b1befa16fae8e95ef50611740d97d4e805f870
Signed-off-by: Uri Eliyahu <uriel@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/4898
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Dmitri Epshtein <dima@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../mv_drivers_lsp/mv_pp2/net_dev/mv_netdev.c      |    2 -
 .../mv_drivers_lsp/mv_pp2/prs/prs_high_sysfs.c     |   19 +++++++++--
 .../mv_drivers_lsp/mv_pp2/prs/prs_sysfs.h          |   34 ++++++++++++++++++++
 3 files changed, 50 insertions(+), 5 deletions(-)
 create mode 100644 arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/prs/prs_sysfs.h

diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/net_dev/mv_netdev.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/net_dev/mv_netdev.c
index f25f124..b4307fb 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/net_dev/mv_netdev.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/net_dev/mv_netdev.c
@@ -3612,7 +3612,6 @@ static void mv_eth_sysfs_exit(void)
 	mv_pp2_cls3_sysfs_exit(&pd->kobj);
 	mv_pp2_cls2_sysfs_exit(&pd->kobj);
 	mv_pp2_cls_sysfs_exit(&pd->kobj);
-	mv_pp2_prs_low_sysfs_exit(&pd->kobj);
 	mv_pp2_prs_high_sysfs_exit(&pd->kobj);
 	mv_pp2_gbe_sysfs_exit(&pd->kobj);
 	/* can't delete, we call to init/clean function from this sysfs */
@@ -3637,7 +3636,6 @@ static int mv_eth_sysfs_init(void)
 	}
 
 	mv_pp2_gbe_sysfs_init(&pd->kobj);
-	mv_pp2_prs_low_sysfs_init(&pd->kobj);
 	mv_pp2_prs_high_sysfs_init(&pd->kobj);
 	mv_pp2_cls_sysfs_init(&pd->kobj);
 	mv_pp2_cls2_sysfs_init(&pd->kobj);
diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/prs/prs_high_sysfs.c b/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/prs/prs_high_sysfs.c
index c75e4b9..6e20205 100644
--- a/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/prs/prs_high_sysfs.c
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/prs/prs_high_sysfs.c
@@ -35,7 +35,9 @@ disclaimer.
 #include "mvCommon.h"
 #include "prs/mvPp2PrsHw.h"
 #include "prs/mvPp2Prs.h"
+#include "prs_sysfs.h"
 
+static struct kobject *prs_kobj;
 
 static ssize_t mv_prs_high_help(char *b)
 {
@@ -206,7 +208,6 @@ static struct attribute *prs_high_attrs[] = {
 };
 
 static struct attribute_group prs_high_group = {
-	.name = "prsHigh",
 	.attrs = prs_high_attrs,
 };
 
@@ -214,15 +215,27 @@ int mv_pp2_prs_high_sysfs_init(struct kobject *pp2_kobj)
 {
 	int err;
 
-	err = sysfs_create_group(pp2_kobj, &prs_high_group);
+	prs_kobj = kobject_create_and_add("prs", pp2_kobj);
+
+	if (!prs_kobj) {
+		pr_err("%s: cannot create gbe kobject\n", __func__);
+		return -ENOMEM;
+	}
+
+	err = sysfs_create_group(prs_kobj, &prs_high_group);
 	if (err)
-		pr_err("sysfs group %s failed %d\n", prs_high_group.name, err);
+		pr_err("sysfs group failed %d\n", err);
+
+	mv_pp2_prs_low_sysfs_init(prs_kobj);
 
 	return err;
+
 }
 
 int mv_pp2_prs_high_sysfs_exit(struct kobject *pp2_kobj)
 {
+	mv_pp2_prs_low_sysfs_exit(prs_kobj);
+
 	sysfs_remove_group(pp2_kobj, &prs_high_group);
 
 	return 0;
diff --git a/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/prs/prs_sysfs.h b/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/prs/prs_sysfs.h
new file mode 100644
index 0000000..503b62e
--- /dev/null
+++ b/arch/arm/plat-armada/mv_drivers_lsp/mv_pp2/prs/prs_sysfs.h
@@ -0,0 +1,34 @@
+/*******************************************************************************
+Copyright (C) Marvell International Ltd. and its affiliates
+
+This software file (the "File") is owned and distributed by Marvell
+International Ltd. and/or its affiliates ("Marvell") under the following
+alternative licensing terms.  Once you have made an election to distribute the
+File under one of the following license alternatives, please (i) delete this
+introductory statement regarding license alternatives, (ii) delete the two
+license alternatives that you have not elected to use and (iii) preserve the
+Marvell copyright notice above.
+
+
+********************************************************************************
+Marvell GPL License Option
+
+If you received this File from Marvell, you may opt to use, redistribute and/or
+modify this File in accordance with the terms and conditions of the General
+Public License Version 2, June 1991 (the "GPL License"), a copy of which is
+available along with the File in the license.txt file or by writing to the Free
+Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 or
+on the worldwide web at http://www.gnu.org/licenses/gpl.txt.
+
+THE FILE IS DISTRIBUTED AS-IS, WITHOUT WARRANTY OF ANY KIND, AND THE IMPLIED
+WARRANTIES OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE ARE EXPRESSLY
+DISCLAIMED.  The GPL License provides additional details about this warranty
+disclaimer.
+*******************************************************************************/
+#ifndef __prs_sysfs_h__
+#define __prs_sysfs_h__
+
+int mv_pp2_prs_low_sysfs_init(struct kobject *kobj);
+int mv_pp2_prs_low_sysfs_exit(struct kobject *kobj);
+
+#endif /* __prs_sysfs_h__*/
-- 
1.7.5.4

